src/CoreBundle/Bridge/Chameleon/Module/NavigationTreeSingleSelect/NavigationTreeSingleSelect.php

```diff
@@ -50,34 +89,69 @@ class NavigationTreeSingleSelect extends MTPkgViewRendererAbstractModuleMapper
      */
     protected $restrictedNodes = [];
 
+    /**
+     * @var TTools
+     */
+    private $tools;
+
     /**
      * @var TGlobal
      */
     private $global;
 
+    /**
+     * @var FieldTranslationUtil
+     */
+    private $fieldTranslationUtil;
+
+    /**
+     * @var \TdbCmsLanguage
+     */
+    private $editLanguage;
+
     public function __construct(
         Connection $dbConnection,
         EventDispatcherInterface $eventDispatcher,
         InputFilterUtilInterface $inputFilterUtil,
+        UrlUtil $urlUtil,
         BackendTreeNodeFactory $backendTreeNodeFactory,
-        TGlobal $global
+        TranslatorInterface $translator,
+        TTools $tools,
+        TGlobal $global,
+        FieldTranslationUtil $fieldTranslationUtil,
+        LanguageServiceInterface $languageService
     ) {
         $this->dbConnection = $dbConnection;
         $this->eventDispatcher = $eventDispatcher;
         $this->inputFilterUtil = $inputFilterUtil;
+        $this->urlUtil = $urlUtil;
         $this->backendTreeNodeFactory = $backendTreeNodeFactory;
+        $this->translator = $translator;
+        $this->tools = $tools;
         $this->global = $global;
+        $this->fieldTranslationUtil = $fieldTranslationUtil;
+
+        $this->editLanguage = $languageService->getActiveEditLanguage();
     }
 
     /**
      * {@inheritdoc}
      */
     public function Accept(\IMapperVisitorRestricted $visitor, $cachingEnabled, \IMapperCacheTriggerRestricted $cacheTriggerManager)
     {
-        $fieldName = $this->inputFilterUtil->getFilteredGetInput('fieldName', '');
-        $activeNodeId = $this->inputFilterUtil->getFilteredGetInput('id', '');
-        $portalSelect = $this->inputFilterUtil->getFilteredGetInput('portalSelect', 0);
-        $isPortalSelectMode = $portalSelect === '1' ? true : false;
+        $this->fieldName = $this->inputFilterUtil->getFilteredGetInput('fieldName', '');
+        $this->activeNodeId = $this->inputFilterUtil->getFilteredGetInput('id', '');
+        $portalSelectMode = $this->inputFilterUtil->getFilteredGetInput('portalSelectMode', '');
+        $visitor->SetMappedValue('portalSelectMode', $portalSelectMode);
+
+        $pagedef = $this->inputFilterUtil->getFilteredGetInput('pagedef', 'navigationTreeSingleSelect');
+        if ('' !== $portalSelectMode) {
```

Comment:
That should be a boolean (as it was).

Reply:
$portalSelectMode can be empty (if it called from a page) or with content (if it called from portal record). Possible values are  "portalSelect" or "portalHomePage". Depending on the value different selections are possible.